lib/webidl2.js

```diff
@@ -124,25 +174,22 @@
       if (t.value.length > 0) {
         if (store) {
           let w = t.value;
-          const re = {
-            "ws": /^([\t\n\r ]+)/,
-            "line-comment": /^\/\/(.*)\r?\n?/,
-            "multiline-comment": /^\/\*((?:.|\n|\r)*?)\*\//
-          };
-          const wsTypes = [];
-          for (const k in re) wsTypes.push(k);
-          while (w.length) {
+          let lastIndex = 0;
+          while (lastIndex < w.length) {
             let matched = false;
-            for (const type of wsTypes) {
-              w = w.replace(re[type], (tok, m1) => {
-                store.push({ type: type + (pea ? ("-" + pea) : ""), value: m1 });
+            for (var type in all_ws_re) {
```

Comment:
Are we going back to for-var-in pattern on purpose? I'm not sure we should...

Reply:
According to https://github.com/w3c/web-platform-tests/commit/280005efe04503f14891d73c6ceab1077a59a735 Servo does not support "const type of" yet. While we don't strictly need to care about it here, if we use it it will have to be patched downstream again.

Reply:
https://github.com/servo/servo/issues/19535

Servo supports `for (let type of foo)` (but in a wrong way), I think keeping for-of syntax should be preferred.

Alternatively the repo may just use a transpiled file. Should we provide one?

Reply:
Maybe we should meet them half-way and use `let`... as there is only one `type` in scope, servo should not get confused, right?

Reply:
I changed it to `let type in all_ws_re`. With the way the code is now it doesn't need `const ... of` anyway.

Reply:
I locally confirmed the latest nightly build has no issue with callback-less synchronous `for (let type of array)`, can we use it so that we can keep `of`?

Reply:
Additionally, a relevant comment will be great so that we won't accidentally do a rollback.

Reply:
We can't use `of` here unless we cache an array of the keys of `all_ws_re`. Is there a benefit in doing that?

I have added a comment about why we are not using `const`.

Reply:
Ah, I just found that `wsTypes` is removed. Never mind, sorry.

I'd like to add a relevant Servo issue URL so that a future contributor can easily follow it, but I couldn't find one. Should I file a new issue?

Reply:
If you have a running copy of Servo to repro on, that would be helpful, yes.